mta/mta_services.go

```diff
@@ -83,71 +91,71 @@ func DeleteMta(path string) error {
-	mta, err := getMtaFromFile(filepath.Join(path))
+func AddModule(path string, moduleDataJSON string, marshal func(*MTA) ([]byte, error)) ([]string, error) {
+	mta, messages, err := GetMtaFromFile(filepath.Join(path), nil)
 	if err != nil {
-		return err
+		return messages, err
 	}
 
 	module := Module{}
 	err = unmarshalData(moduleDataJSON, &module)
 	if err != nil {
-		return err
+		return messages, err
 	}
 
 	mta.Modules = append(mta.Modules, &module)
-	return saveMTA(path, mta, marshal)
+	return messages, saveMTA(path, mta, marshal)
 }
 
 //AddResource - adds a new resource.
-func AddResource(path string, resourceDataJSON string, marshal func(*MTA) ([]byte, error)) error {
-	mta, err := getMtaFromFile(path)
+func AddResource(path string, resourceDataJSON string, marshal func(*MTA) ([]byte, error)) ([]string, error) {
+	mta, messages, err := GetMtaFromFile(path, nil)
 	if err != nil {
-		return err
+		return messages, err
 	}
 
 	resource := Resource{}
 	err = unmarshalData(resourceDataJSON, &resource)
 	if err != nil {
-		return err
+		return messages, err
 	}
 
 	mta.Resources = append(mta.Resources, &resource)
-	return saveMTA(path, mta, marshal)
+	return messages, saveMTA(path, mta, marshal)
 }
 
 //GetModules - gets all modules.
-func GetModules(path string) ([]*Module, error) {
-	mta, err := getMtaFromFile(path)
+func GetModules(path string, extensions []string) ([]*Module, []string, error) {
+	mta, messages, err := GetMtaFromFile(path, extensions)
 	if err != nil {
-		return nil, err
+		return nil, messages, err
 	}
-	return mta.Modules, nil
+	return mta.Modules, messages, nil
 }
 
 //GetResources - gets all resources.
-func GetResources(path string) ([]*Resource, error) {
-	mta, err := getMtaFromFile(path)
+func GetResources(path string, extensions []string) ([]*Resource, []string, error) {
+	mta, messages, err := GetMtaFromFile(path, extensions)
 	if err != nil {
-		return nil, err
+		return nil, messages, err
 	}
-	return mta.Resources, nil
+	return mta.Resources, messages, nil
 }
 
 // GetResourceConfig returns the configuration for a resource (its service creation parameters).
 // If both the config and path parameters are defined, the result is merged.
-func GetResourceConfig(path string, resourceName string, workspaceDir string) (map[string]interface{}, error) {
-	mta, err := getMtaFromFile(path)
+func GetResourceConfig(path string, extensions []string, resourceName string, workspaceDir string) (map[string]interface{}, []string, error) {
+	mta, messages, err := GetMtaFromFile(path, extensions)
 	if err != nil {
-		return nil, err
+		return nil, messages, err
 	}
 	if len(workspaceDir) == 0 {
 		workspaceDir = filepath.Dir(path)
 	}
 
 	resource := mta.GetResourceByName(resourceName)
 	if resource == nil {
-		return nil, fmt.Errorf("the '%s' resource does not exist", resourceName)
+		return nil, messages, fmt.Errorf("the '%s' resource does not exist", resourceName)
```

Comment:
in most places we use like: `the "%s" resource does not exist`

Reply:
I didn't change the error message in this PR so I think I will leave it like this